src/events/api/views.py

```diff
@@ -16,9 +17,14 @@ class TalkDetailAPIView(RetrieveAPIView):
     authentication_classes = [TokenAuthentication]
     permission_classes = [IsAuthenticated]
 
-    queryset = TalkProposal.objects.all()
+    queryset = ProposedTalkEvent.objects.all()
     serializer_class = serializers.TalkDetailSerializer
 
+    def get_object(self):
```

Comment:
@setmao I just realized that these lines need to be kept since the `pk` passed-in is actually the pk of `TalkProposal` and the function return the associated `ProposedTalkEvent` with `get_object_or_404(queryset, proposal_id=pk)`